frontend/src/components/Map/LayerDrawer.tsx

```diff
@@ -47,10 +52,36 @@ const drawerStyles = css`
   border-bottom-right-radius: 2px 2px;
 `;
 
+const dragIndicatorStyle = css`
+  color: #aaa;
+  :hover {
+    cursor: grab;
+    color: #777;
+    transform: scale(1.3);
+  }
+  transition: transform 0.1s ease-in-out;
+`;
+
+const toggleOnStyle = css`
+  color: green;
+`;
+
+const toggleOffStyle = css`
+  color: #aaa;
+`;
+
+function setLayerSelected(layersState: LayerState[], layerId: VectorLayerKey, selected: boolean) {
+  const newLayers = layersState.slice();
+  const layerIndex = newLayers.findIndex((layer) => layer.id === layerId);
+  newLayers[layerIndex] = { ...newLayers[layerIndex], selected };
+  return newLayers;
```

Comment:
[nit] I think this could be written with a single `map` function
```suggestion
  return layersState.map((layer) =>
    layer.id === layerId
      ? {
          ...layer,
          selected,
        }
      : layer
  );
```